controllers/clustergroupupgrade_controller.go

```diff
@@ -361,43 +341,87 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 			nextReconcile = requeueWithMediumInterval()
 		} else {
 			// There are no blocking CRs, continue with the upgrade process.
-			// Take actions before starting upgrade.
-			err = r.takeActionsBeforeEnable(ctx, clusterGroupUpgrade)
+			// create backup
+			err = r.reconcileBackup(ctx, clusterGroupUpgrade)
 			if err != nil {
+				r.Log.Error(err, "reconcileBackup error")
 				return
 			}
 
-			// If the remediation plan is empty, update the status.
-			if clusterGroupUpgrade.Status.RemediationPlan == nil {
-				utils.SetStatusCondition(
-					&clusterGroupUpgrade.Status.Conditions,
-					utils.ConditionTypes.Progressing,
-					utils.ConditionReasons.Completed,
-					metav1.ConditionFalse,
-					"The ClusterGroupUpgrade CR has all clusters compliant with all the managed policies",
-				)
-				utils.SetStatusCondition(
-					&clusterGroupUpgrade.Status.Conditions,
-					utils.ConditionTypes.Succeeded,
-					utils.ConditionReasons.UpgradeCompleted,
-					metav1.ConditionTrue,
-					"The ClusterGroupUpgrade CR has all clusters already compliant with the specified managed policies",
-				)
-				nextReconcile = requeueImmediately()
-			} else {
-				// Start the upgrade.
-				utils.SetStatusCondition(
-					&clusterGroupUpgrade.Status.Conditions,
-					utils.ConditionTypes.Progressing,
-					utils.ConditionReasons.InProgress,
-					metav1.ConditionTrue,
-					"The ClusterGroupUpgrade CR has upgrade policies that are still non compliant",
-				)
-				clusterGroupUpgrade.Status.Status.StartedAt = metav1.Now()
-				nextReconcile = requeueImmediately()
+			if clusterGroupUpgrade.Status.Backup != nil {
+				for _, v := range clusterGroupUpgrade.Status.Backup.Status {
+					//nolint
+					if v == BackupStatePreparingToStart || v == BackupStateStarting || v == BackupStateActive {
+						utils.SetStatusCondition(
+							&clusterGroupUpgrade.Status.Conditions,
+							utils.ConditionTypes.Progressing,
+							utils.ConditionReasons.NotStarted,
+							metav1.ConditionFalse,
+							"The Cluster backup is in progress",
+						)
+						err = r.updateStatus(ctx, clusterGroupUpgrade)
+						nextReconcile = requeueWithShortInterval()
+						return
+					}
+				}
 			}
-		}
 
+			backupCondition := meta.FindStatusCondition(clusterGroupUpgrade.Status.Conditions, string(utils.ConditionTypes.BackupSuceeded))
+			if clusterGroupUpgrade.Status.Backup == nil || (backupCondition != nil && backupCondition.Status == metav1.ConditionTrue) {
+
+				// rebuild remediation plan, since backup has finished.
+				if clusterGroupUpgrade.Status.Backup != nil {
+					var allManagedPoliciesExist bool
```

Comment:
We could avoid this duplicated block if we requeue after backup is done.

Reply:
added a check [44b6fc1](https://github.com/openshift-kni/cluster-group-upgrades-operator/pull/335/commits/44b6fc11b1d841dcb22459ed81178cc674de4774), it should avoid the block after progressing is set to true

Reply:
this is not what I meant and it seems to break integration tests. you can put it back and I will merge it as soon as tests pass. we will address the comments in a followup PR.